### PR TITLE
Make sure MinMaxCloseControl supports fullscreen and tablet mode

### LINKED ISSUE #2001: Make sure MinMaxCloseControl supports FullScreen & Tablet Mode
Some thoughts about the Min/Max/Close buttons, and supporting FullScreen or Tablet Mode

Some properties to consider for this TitlebarControl.  This could be also be useful for the Settings UI, if it is presented in a window.
* CanMaximise
* CanMinimise
* CanRestore

Which would hide those buttons if you wanted to make a settings screen which only shows a close button.  Also consider how these TitleBar Controls will handle FullScreen/Tablet Mode

image
_Full Screen_

Screenshot (17)
_Tablet Mode_

_Originally posted by @mdtauk in https://github.com/microsoft/terminal/pull/1948#issuecomment-512048300_

### PR DESCRIPTION
**Pull request by AnuthaDev**
## Summary of the Pull Request
Caption buttons support fullscreen and tablet mode
 
## References
https://devblogs.microsoft.com/oldnewthing/?p=93815
https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/move-to-winrt-from-wrl
https://devblogs.microsoft.com/oldnewthing/20170405-00/?p=95905

## PR Checklist
* [x] Closes #2001
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
Used [this](https://devblogs.microsoft.com/oldnewthing/?p=93815) to add support for detecting tablet mode

## Validation Steps Performed
None
You know how it is, coded on phone, can't get it to compile 😔

### CODE REVIEW

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp [hunk 1]
```diff
@@ -788,6 +810,15 @@ void NonClientIslandWindow::_UpdateFrameMargins() const noexcept
         return FALSE;
     }
 
+    WRL::ComPtr<IUIViewSettingsInterop> interop;
+    Windows::Foundation::GetActivationFactory(WRL::Wrappers::HStringReference(
```
Comment (AnuthaDev) [inline src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp:814]: @DHowett Please tell me how to convert it from WRL to winrt🙏🙏

Comment (DHowett) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp:814]: So, to convert this from WRL to WinRT, all you need to do is.. replace all this code with:

```
auto viewSettingsForWindow = winrt::Windows::UI::ViewManagement::UIViewSettings::GetForWindow();
```

C++/WinRT makes winrt look like modern C++, so you don't need to do any weird smart pointer types or IID management.

Comment (DHowett) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp:814]: Oh, but if you have to use an Interop class... then things get weird and annoying. You actually can't use c++/winrt directly for this. lol ☹️

Comment (AnuthaDev) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp:814]: Well, using an interop class is the only way I know of. Summoning the legend himself for guidance : @oldnewthing

Comment (oldnewthing) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp:814]: Oh you mean

```cpp
    UIViewSettings viewSettingsForWindow{ nullptr };
    get_activation_factory<IUIViewSettingsInterop>(name_of<UIViewSettings>())->
        GetForWindow(hwnd, guid_of<UIViewSettings>(), put_abi(viewSettingsForWindow));
```

or if you like one-liners

```cpp
    auto viewSettingsForWindow =
        capture<IInspectable>(
            get_activation_factory<IUIViewSettingsInterop>(name_of<UIViewSettings>()),
            &IUIViewSettingsInterop::GetForWindow,
            hwnd).as<UIViewSettings>();
```

**Comment by zadjii-msft** [comment]
Hey this one's been sitting open for quite some time now - @AnuthaDev any plans to come back around and finish this one?

**Pull request closed without merging**